gprofiler/merge.py

```diff
@@ -434,12 +435,22 @@ def merge_profiles(
         profile_samples_count = sum(profile.stacks.values())
         assert profile_samples_count > 0
 
-        # do the scaling by the ratio of samples: samples we received from perf for this process,
-        # divided by samples we received from the runtime profiler of this process.
-        ratio = perf_samples_count / profile_samples_count
-        profile.stacks = scale_sample_counts(profile.stacks, ratio)
-
-        # swap them: use the samples from the runtime profiler.
+        if process_perf is not None and perf_samples_count > 0 and ProfilingErrorStack.is_error_stack(profile.stacks):
+            # runtime profiler returned an error stack; extend it with perf profiler stacks for the pid
+            profile.stacks = ProfilingErrorStack.attach_error_to_stacks(process_perf.stacks, profile.stacks)
+            metadata = dict(profiling_mode="cpu")
+            enrichment_options = EnrichmentOptions(
+                profile_api_version=None,
+                container_names=False,
+                application_identifiers=False,
+                application_identifier_args_filters=[],
+                application_metadata=False,
+            )
+        else:
+            # do the scaling by the ratio of samples: samples we received from perf for this process,
+            # divided by samples we received from the runtime profiler of this process.
+            ratio = perf_samples_count / profile_samples_count
+            profile.stacks = scale_sample_counts(profile.stacks, ratio)
+        # swap them: use the processed (scaled or extended) samples from the runtime profiler.
```

Comment:
Can also keep the one blank line between the `if` and this line, as there was before a blank between the 2 blocks

Reply:
👍